Keen.NetStandard/EventCachePortable.cs

```diff
@@ -0,0 +1,201 @@
+﻿using Keen.NetStandard.EventCache;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using System.IO;
+using System.Text;
+
+namespace Keen.NetStandard
+{
+    /// <summary>
+    /// <para>EventCachePortable implements the IEventCache interface using
+    /// file-based storage. It has no cache-expiration policy.</para>
+    /// <para>To use, pass an instance of this class when constructing KeenClient.
+    /// To construct a new instance, call the static New() method.</para>
+    /// </summary>
+    public class EventCachePortable : IEventCache
+    {
+        // An asynchronous, lazy loaded instance of this class.
+        private static AsyncLazy<EventCachePortable> _instanceTask =
+            new AsyncLazy<EventCachePortable>(
+                taskFactory: CreateInstanceAsync);
+
+        // A list of events in the cache
+        protected List<string> events = new List<string>();
+
+        protected EventCachePortable() { }
+
+        /// <summary>
+        /// Get the singleton instance of EventCachePortable.
+        /// Since initialization is async, this wraps the instance
+        /// with a Task.
+        /// </summary>
+        /// <para>To get the EventCachePortable instance, await this property.</para>
+        /// <para>The Task<EventCachePortable> is available via Value on this property.</para>
+        /// <returns></returns>
+        public static AsyncLazy<EventCachePortable> InstanceTask => _instanceTask;
+
+        /// <summary>
+        /// Create, initialize and return an instance of EventCachePortable.
+        /// </summary>
+        /// <returns></returns>
+        private static async Task<EventCachePortable> CreateInstanceAsync()
+        {
+            var cache = new EventCachePortable();
+
+            await cache.InitializeAsync();
+
+            return cache;
+        }
+
+        /// <summary>
+        /// Initialize an instance by ensuring the cache path exists,
+        /// then read all existing cached events into the in-memory queue.
+        /// </summary>
+        /// <returns></returns>
+        protected async Task InitializeAsync()
+        {
+            // Get/create the cache directory
+            DirectoryInfo keenFolder = await GetOrCreateKeenDirectoryAsync()
+                .ConfigureAwait(continueOnCapturedContext: false);
+
+            // Read all files from the cache. Each one contains a json
+            // serialized CachedEvent
```

Comment:
Can we file an Issue to make this more like the Java SDK and have a per-collection file if that makes sense. The issue there of course becomes whether or not the Java SDK is doing the right thing in terms of accessing as per-collection file in a thread-safe way from different KeenClient instances.